src/libstd/sys/unix/net.rs

```diff
@@ -66,7 +67,30 @@ impl Socket {
     }
 
     pub fn duplicate(&self) -> io::Result<Socket> {
-        let fd = try!(cvt(unsafe { libc::dup(self.0.raw()) }));
+        use libc::funcs::posix88::fcntl::fcntl;
+        const ZERO: c_int = 0;
```

Comment:
I can't guarantee it's `c_int` then, because the compiler doesn't (/can't) know the type of the third parameter. Whether anything goes wrong or not, one would need to have better understanding of this situation than I do because this is unsafe code.

(For reference, the function signature is declared as `int fcntl(int filedes, int cmd, ...)`, and in our case `int fcntl(int filedes, F_DUPFD_CLOEXEC, int arg)`.)